src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -544,32 +565,47 @@ public TranslogStats stats() {
         }
     }
 
-    private boolean isReferencedTranslogId(long translogId) {
-        return translogId >= lastCommittedTranslogId;
+    private boolean isReferencedGeneration(long generation) { // pkg private for testing
+        return generation >= lastCommittedTranslogFileGeneration;
     }
 
-    private final class InternalChannelReference extends ChannelReference {
-        final long translogId;
+    public TranslogConfig getConfig() {
+        return config;
+    }
 
-        public InternalChannelReference(long translogId, Path file, OpenOption... openOptions) throws IOException {
-            super(file, openOptions);
-            this.translogId = translogId;
```

Comment:
Hmm the comment is wrong?  It's private not pkg private, and it's used "for real" not just for testing?